src/builder/PaycallWrapper.sol

```diff
@@ -12,15 +13,19 @@ library PaycallWrapper {
         pure
         returns (IQuarkWallet.QuarkOperation memory)
     {
+        PaymentTokens.PaymentToken memory paymentToken = PaymentTokens.knownToken(chainId);
+        bytes memory paycallSource =
+            abi.encodePacked(type(Paycall).creationCode, abi.encode(paymentToken.priceFeed, paymentToken.token));
         bytes[] memory scriptSources = new bytes[](operation.scriptSources.length + 1);
         for (uint256 i = 0; i < operation.scriptSources.length; i++) {
             scriptSources[i] = operation.scriptSources[i];
         }
-        scriptSources[operation.scriptSources.length] = type(Paycall).creationCode;
+
```

Comment:
nit: extra line

Reply:
oh haha.  I'm used to this convention that always leave a extra line after loop `}` for readability 😄